test/schema-special-usage.test.js

```diff
@@ -47,6 +50,122 @@ test('Ajv8 usage instead of the bundle one', t => {
   })
 })
 
+test('JTD Usage via manual inclusion', t => {
+  t.plan(2)
+
+  t.test('use JTD explicit substitution to fail on JSON Schema', t => {
+    t.plan(2)
+    const ajv = new AjvJTD({
+      allErrors: true
+    })
+    const fastify = Fastify({ jsonShorthand: false })
+    fastify.setValidatorCompiler(({ schema }) => {
+      return ajv.compile(schema)
+    })
+
+    fastify.post('/', {
+      schema: {
+        body: {
+          // JSON Schema style discriminator is invalid in JTD schemas
+          discriminator: { propertyName: 'version' },
+          oneOf: [
+            {
+              properties: {
+                version: { const: '1' },
+                foo: { type: 'string' }
+              },
+              required: ['version']
+            },
+            {
+              properties: {
+                version: { const: '2' },
+                foo: { type: 'number' }
+              },
+              required: ['version']
+            }
+          ]
+        }
+      },
+      handler (req, reply) { reply.send({ ok: 1 }) }
+    })
+
+    fastify
+      .ready(err => {
+        t.ok(err)
+        t.match(
+          err ? err.message : undefined,
```

Comment:
this cheks is not useful and fagile because it is based on an external module out of our control

Reply:
I agree, I wanted to have a test that implemented JTD, but now I'm thinking it is indeed a bit misplaced since this PR isn't specifically adding JTD support anyway, testing one of many possible use cases probably isn't the right test to have there.